src/main/java/mate/academy/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package mate.academy.service.impl;
+
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import mate.academy.dto.user.UserRegistrationRequest;
+import mate.academy.dto.user.UserResponseDto;
+import mate.academy.exception.EntityNotFoundException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.mapper.UserMapper;
+import mate.academy.model.User;
+import mate.academy.repository.role.RoleRepository;
+import mate.academy.repository.user.UserRepository;
+import mate.academy.service.UserService;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private static final Long USER_ROLE_ID = 2L;
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+    private final PasswordEncoder passwordEncoder;
+    private final RoleRepository roleRepository;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequest request) throws RegistrationException {
+        if (userRepository.findByEmail(request.getEmail()).isPresent()) {
+            throw new RegistrationException("User already exists");
+        }
+        User user = new User();
+        user.setPassword(passwordEncoder.encode(request.getPassword()));
+        user.setEmail(request.getEmail());
+        user.setRoles(Set.of(roleRepository.findById(USER_ROLE_ID)
```

Comment:
To my mind, It is better to search by the "roleName" field.
```suggestion
        user.setRoles(Set.of(roleRepository.findByRoleName(RoleName.USER)
```